PetaStarter/Areas/SBBoss/Controllers/FinanceInfoController.cs

```diff
@@ -45,11 +46,43 @@ public ActionResult Index(PaymentViewModel paymentView, int? mode)
                     };
                     db.Insert(payment);
 
+                    string note = null;
+                    string driver = null;
+                    string supplier = null;
+                    string agent = null;
+                    if (payment.DriverID != null)
+                    {
+                        driver = db.SingleOrDefault<Driver>(paymentView.DriverID).DriverName;
+                    }
+                    if (payment.SupplierID != null)
+                    {
+                        supplier = db.SingleOrDefault<Driver>(paymentView.DriverID).DriverName;
+                    }
+                    if(payment.AgentId != null)
+                    {
+                        agent = db.ExecuteScalar<string>("select ContactName from Agent where AgentId = @0", paymentView.AgentId);
+                    }
+
+                    if (mode == (int)PayToEnum.Driver)
+                    {
+                        note = $"Paid to {driver} Rs. {paymentView.Amount} towards settlement of bookings ";
+                    }
+                    if (mode == (int)PayToEnum.Supplier)
+                    {
+                        note = $"Paid to {supplier} Rs. {paymentView.Amount} towards settlement of bookings ";
+                    }
+                    if (mode == (int)PayToEnum.Agent)
+                    {
+                        note = $"Recieved from {agent} Rs. {paymentView.Amount} towards settlement of bookings ";
+                    }
+
```

Comment:
Please remember to also add the note that the user has entered in the Finance screen.
note= note+UserNote;

Reply:
note= note+" User note: " + UserNote;